km/km_cpu_init.c

```diff
@@ -283,6 +283,9 @@ km_vcpu_t* km_vcpu_get(void)
 {
    km_vcpu_t* vcpu;
 
+   if (machine.exit_group != 0) {
```

Comment:
Consider current thread crossing this line and getting scheduled out other thread then runs exit group  and exit_grp run's farther before first thread gets scheduled back.

Reply:
You right.